kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+#include <errno.h>
+
+#ifndef _PTHREAD_SEMAPHORE_H
+#define _PTHREAD_SEMAPHORE_H
+
+int sem_destroy(sem_t *semaphore)
+{
+	if (k_sem_count_get(semaphore->sem) != 0) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	k_sem_reset(semaphore->sem);
+
+	return 0;
+}
+
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	*value = (int) k_sem_count_get(semaphore->sem);
+
+	return 0;
+}
+
+int sem_init(sem_t *semaphore, int pshared, unsigned int value)
+{
+	if ((semaphore->sem->limit) == 0) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	__ASSERT((pshared == 1) || (pshared == 0), "not correct pshared");
+
+	k_sem_init(semaphore->sem, value, semaphore->sem->limit);
+
+	return 0;
+}
+
+int sem_post(sem_t *semaphore)
+{
+	if (semaphore->sem == 0) {
```

Comment:
Why is a pointer being compared to `0`? Can this simply be a call to `k_sem_give()`?